src/input_processing/dual_axis.rs

```diff
@@ -0,0 +1,1544 @@
+///
+/// ```rust
+/// use bevy::prelude::*;
+/// use leafwing_input_manager::prelude::*;
+///
+/// let exclusion = CircleExclusion::new(0.3);
+/// let bounds = CircleBounds::new(2.4);
+/// let deadzone = CircleDeadzone::new(exclusion, bounds);
+///
+/// // Another way to create a CircleDeadzone.
+/// let alternative = exclusion.normalized_into(bounds);
+/// assert_eq!(alternative, deadzone);
+///
+/// // These values have a magnitude within the radius of the exclusion.
+/// let values = [Vec2::new(0.0, 0.2), Vec2::new(0.1, 0.15)];
+/// for value in values {
+///     assert!(value.length() <= exclusion.radius());
+///
+///     // So the value should be excluded.
+///     assert!(exclusion.contains(value));
+///
+///     // So the value should be treated as zeros.
+///     let result = deadzone.process(value);
+///     assert_eq!(result, Vec2::ZERO);
+/// }
+///
+/// // The values have a magnitude at or exceed the maximum bound.
+/// let values = [Vec2::new(2.0, 10.0), Vec2::splat(5.0)];
+/// for value in values {
+///     assert!(value.length() >= bounds.radius());
+///
+///     // So the value is out of the bounds.
+///     assert!(!bounds.contains(value));
+///
+///     // So the value should be clamped to the maximum bound.
+///     let result = deadzone.process(value);
+///     assert_eq!(result.length(), bounds.radius());
+///     assert_eq!(result.y.atan2(result.x), value.y.atan2(value.x));
+/// }
+///
+/// // These values are within the livezones.
+/// let values = [Vec2::new(0.9, -1.5), Vec2::new(-1.3, 1.0)];
+/// for value in values {
+///     assert!(value.length() > exclusion.radius());
+///     assert!(value.length() < bounds.radius());
+///
+///     // So the value should be normalized to fit the range.
+///     let result = deadzone.process(value);
+///     assert!(result.length() > 0.0);
+///     assert!(result.length() < bounds.radius());
+///     assert_eq!(result.y.atan2(result.x), value.y.atan2(value.x));
+/// }
+/// ```
+#[derive(Debug, Clone, Copy, PartialEq, Reflect, Serialize, Deserialize)]
+#[must_use]
+pub struct CircleDeadzone {
+    /// The exclusion used for normalization.
+    pub(crate) exclusion: CircleExclusion,
+
+    /// The bounds used for normalization.
+    pub(crate) bounds: CircleBounds,
+
+    /// Pre-calculated reciprocal of the livezone radius,
+    /// preventing division during normalization.
+    pub(crate) livezone_recip: f32,
+}
+
+#[typetag::serde]
+impl DualAxisProcessor for CircleDeadzone {
+    /// Normalizes all input values within the livezone regions and returns the result.
+    #[must_use]
+    fn process(&self, input_value: Vec2) -> Vec2 {
+        // Short-circuiting if the input value should be excluded.
+        if self.exclusion.contains(input_value) {
+            return Vec2::ZERO;
+        }
+
+        // Short-circuiting if the input value should be clamped.
+        if !self.bounds.contains(input_value) {
+            return self.bounds.process(input_value);
+        }
+
```

Comment:
I reviewed the generated assembly code and observed that the compiler smartly inlined these functions and removed redundant code segments, resulting in an outcome identical to manually inlining the code.